### PR TITLE
Simplified parameter handling

### PR DESCRIPTION
**Pull request by OskarStark**
IMO this is much cleaner!

i added some more tests in a separate commit

This PR replaces #720, because it wasn't done on a forked repo, sorry!

### CODE REVIEW

**Comment by OskarStark** [comment]
ping @greg0ire @Soullivaneuh

File: Tests/Entity/UserManagerTest.php [hunk 1]
```diff
@@ -42,7 +42,8 @@ public function testGetPager()
                     $self->equalTo('u.username'),
                     $self->equalTo('ASC')
                 );
-                $qb->expects($self->once())->method('setParameters')->with($self->equalTo(array()));
+                $qb->expects($self->never())->method('setParameter');
+                $qb->expects($self->never())->method('setParameters');
```
Comment (OskarStark) [inline Tests/Entity/UserManagerTest.php]: not sure if this is needed

Comment (greg0ire) [inline Tests/Entity/UserManagerTest.php]: We should test how the object behaves, not how it is implemented.

Comment (OskarStark) [inline Tests/Entity/UserManagerTest.php]: what do you mean by that?

**Pull request merged**